### PR TITLE
Clarifications to make coverity happy

### PR DESCRIPTION
**Pull request by keszybz**
Check or explicitly ignore various return values and simplify one function to avoid TOCTOU.

### CODE REVIEW

File: src/libsystemd/sd-device/sd-device.c [hunk 1]
```diff
@@ -1873,13 +1872,33 @@ _public_ int sd_device_set_sysattr_value(sd_device *device, const char *sysattr,
...
-        if (r < 0) {
+
+        fd = open(path, O_WRONLY | O_CLOEXEC | O_NOFOLLOW);
+        if (fd < 0) {
+                if (errno == ELOOP)
+                        return -EINVAL;
+                if (errno == EISDIR)
+                        return -EISDIR;
+
                 value = strdup("");
```
Comment (martinpitt) [inline src/libsystemd/sd-device/sd-device.c:1888]: This clobbers the strdup'ed value from above (line 1885) and thus leaks memory. Maybe move the strdup into an else branch of this `if`? Or even easier, do the strdup right before the one call that actually needs it, i. e. `device_add_sysattr_value()`, and drop this and the above strdup?

I also don't quite understand the logic here -- if the `open()` fails, this leaves the real sysfs unchanged, but it calls `device_add_sysattr_value()` with an empty value. Why should the internal state change here if the real state doesn't?

File: src/shared/firewall-util.c [hunk 2]
```diff
@@ -76,8 +76,11 @@ static int entry_fill_basics(
         }
 
         if (out_interface) {
+                unsigned l = strlen(out_interface);
```
Comment (poettering) [inline src/shared/firewall-util.c]: this doesn't look right. strlen() returns `size_t`, not `unsigned` which is 64bit vs 32bit on x86-64...

File: src/libsystemd/sd-device/sd-device.c [hunk 3]
```diff
@@ -1873,13 +1872,33 @@ _public_ int sd_device_set_sysattr_value(sd_device *device, const char *sysattr,
                 return 0;
         }
 
+        value_len = strlen(_value);
+
+        /* drop trailing newlines */
+        while (value_len > 0 && _value[value_len - 1] == '\n')
+                _value[--value_len] = '\0';
```
Comment (poettering) [inline src/libsystemd/sd-device/sd-device.c]: i really don't like that we clobber the input parameter here... This is against our CODING_STYLE really. We should copy this before we chop off newlines, not after

File: src/libsystemd/sd-device/sd-device.c (hunk 1 above)
Comment (keszybz) [reply src/libsystemd/sd-device/sd-device.c:1888]: Beats me ;) I think it's very dubious. I think the idea was that if the attribute does not exist, the internal state should be set to empty. I'm not sure what this function is used for, and I was trying not to change the semantics.

**Comment by keszybz** [comment]
Updated.

**Pull request merged**